### PR TITLE
Monodrag expose min distance to accept pointer

### LINKED ISSUE #82113: Expose `_hasSufficientGlobalDistanceToAccept` in `monodrag.dart`
## Use case

Make gesture recognizers more fine-grained in part of what distance pointer can travel before this will be considered a drag gesture. Particularly, this may allow to fix this TODO

https://github.com/flutter/flutter/blob/29898812d476051c750ae8979e1ad8e212a00f29/packages/flutter/lib/src/gestures/constants.dart#L66-L68

And also I'm considering that we make the Slider to comply with touch gesture immediately after it starts. Currently the hit slop for it is too big and it feels like it lags

## Proposal

Expose `_hasSufficientGlobalDistanceToAccept`

We could also change the way it works to make it similar to `checkForResolutionAfterMove` in `multidrag.dart`

### PR DESCRIPTION
**Pull request by nt4f04uNd**
Allows to configure the drag gesture slop

Fixes https://github.com/flutter/flutter/issues/82113

Along with it, also made a few refactoring changes

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/gestures/monodrag.dart [hunk 1]
```diff
@@ -33,6 +33,12 @@ typedef GestureDragEndCallback = void Function(DragEndDetails details);
 /// Used by [DragGestureRecognizer.onCancel].
 typedef GestureDragCancelCallback = void Function();
 
+/// Signature for a function that computes the distance pointer has to travel
```
Comment (goderbauer) [inline packages/flutter/lib/src/gestures/monodrag.dart:36]: nit: (here and elsewhere) keep doc lines to approx. 80 chars.

**Review by goderbauer — COMMENTED** [review]
What's the motivation for exposing this?

**Comment by nt4f04uNd** [comment]
I listed a few use cases in the issue https://github.com/flutter/flutter/issues/82113 itself
The slider thing I'm talking about there is https://github.com/flutter/flutter/pull/82154

**Comment by goderbauer** [comment]
I am not convinced that https://github.com/flutter/flutter/pull/82154 is a great idea since it doesn't really fix the issue. Until we have an actual use case we should hold off on adding this.

**Pull request closed without merging**